### PR TITLE
Ports bartender drink flinging code updates.

### LINKED ISSUE #10757: Drink Flinging is currently broken.
## Test Merges:

The following pull requests are currently test merged: 
#10541: 'adds the POOL to box station' by Detective-Google at commit 0b119cbd52 
#10564: ' Experimental optimizations' by kevinz000 at commit 388e98bd67 
#10575: '[TESTMERGE] Properly working TGUI Next' by Arturlang at commit 6619ac41c8 
#10649: 'Completely reworks the hijack objective, and shuttle hijacking in general' by kevinz000 at commit 40a2698b55 
#10414: 'The Great Citadel De-Pissening of 2020' by CameronWoof at commit 0c70fc9234 
#10572: 'Removes harm prevention clause indicators from relevant sources.' by Tupinambis at commit dd9e62c8bb 
#10389: '[TESTMERGE ONLY] dumb hotfix' by Putnam3145 at commit 36bc39aad8 
#10581: '[WIP] Taser re-rework: Turns tasers into a right-click action of hybrid tasers, nerfs electrodes similarly to taser kneecapping' by deathride58 at commit b931255abb 

## Reproduction:
The action granted by the book "Tapper: This One's For You" does not work.
Read the book.
Activate the skill.
Glasses still break.

Skill still doesn't work while deactivated.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Ports tgstation PRs #42306 and #48921 by coiax and itseasytosee. The action button toggle was a small sacrifice but it's better this way tbh.

Also fixing a little oversight.

## Why It's Good For The Game
Little update and bartender_check() code improvement. Will close #10757.

## Changelog
:cl: coiax, itseasytosee
tweak: Bartenders now gain their ability to "booze slide" from their
beer googles, rather than from a granter book in their backpacks.
add: Crafting with sunglasses and science goggles now creates science glasses that are identical to beer goggles but don't allow the throwing of drinks without spilling and have a purple tint.
imageadd: Sprite for science glasses.
imageadd: New sprite for beer goggles.
fix: fixed drink flinging / booze sliding.
/:cl:

### CODE REVIEW

**Comment by XyvMeph** [comment]
Hi, Mephistopheles/Xyverious here, and I guess I'll be the first and perhaps only person to complain about this.
A lot of Bartenders that do Bartending don't wear their glasses because they like Aesthetic more than gamer gear. What's more, the whole point of the book is to keep it after we used the book for its purpose. With this, if we lose the glasses, we lose the ability to slide/fling drinks, and frankly the glasses look bad on a lot of characters.
What's more, the only way to actually get the Bartender book beforehand was to steal the book, ask nicely, or order the Build-a-bar from Cargo, which almost no one ever did except for the rare moment we have more than 2 fully functioning bars on station, and most maint bar-runners don't actually get the drink-flinging option, and nor should they, since Bartender is a solo role.

TLDR: Don't do this, because the book is exclusively for the bartender at round-start. Making it more commonplace takes the pizazz out of bartender, and frankly the glasses look like crap on most characters.

**Comment by davidchan13** [comment]
Making it part of their hat, vest, overcoat or gloves would be fine.

The goggles are god damn hideous and I'd much rather have it be a toggeable skill than applied to eyewear, particularly if said eyewear can't even be made to something themed to a bartender like a monocle.

**Comment by Ghommie** [comment]
Oh man, i did not expect people to care about how their 32x32 pixel dudes look far more than actual utilities, moreso for a pair of sunglasses that let em scan reagents, which is pretty useful for the role.
Also fyi there are spare bartender googles in the BarDrobe, whose refills can be ordered from cargo too.

**Comment by XyvMeph** [comment]
> Oh man, i did not expect people to care about how their 32x32 pixel dudes look far more than actual utilities, moreso for a pair of sunglasses that let em scan reagents, which is pretty useful for the role.

Useful for the role, not for aesthetic, and honestly most Bartender mains care way more about that than knowing what's in a drink. They'll put them on to check to stop people from spiking shit, but they rarely, if ever, wear them constantly.

The goggles do not need the ability. The book does.

**Comment by Owai-Seek** [comment]
It's more that you're restricting an ability to equipment on service staff when it wasn't a restriction before, people not wanting to wear glasses for it is just something extra.

It'd be like forcing a janitor to wear galoshes to use a spraybottle, or a botanist to wear gloves to touch any of their plants.

**Comment by Ghommie** [comment]
Uh, screw it.
You snowflakes.
